[Go] Add CreateFollowship handler test that satisfies test isolation

Rwatana opened the pull request:
## Issue Number
#690 

## Topic
Changed NewCreateFollowshipHandler to accept a repository.UsersRepository instead of a raw *sql.DB.
Replaced the old handler.NewCreateFollowshipHandler(db) call with handler.NewCreateFollowshipHandler(infrastructure.InjectUsersRepository(db)) to switch between test and normal situations.
Covered three cases: successful creation, invalid JSON, and use-case error.

## Scope of Impact

- Endpoint affected: POST /api/users/{id}/following

- Handler construction: now uses injected repository rather than direct DB access


## Particular Points to Check
- Modified handler works properly
- Test works individually



## Test
- handler test
success
![image](https://github.com/user-attachments/assets/7b469ab8-acb6-4a96-ba4e-b0bfc2929121)
failure(internal server error)
![image](https://github.com/user-attachments/assets/d969cf7f-3bf5-4b53-992a-922d18aacc08)

failure(bad request body)
![image](https://github.com/user-attachments/assets/28006c0b-83b7-45fc-91af-42e2ef99db17)

- fake handler test
command
```
go test -v ./internal/controller/handler
```
result
```
➜  go git:(feature/#690_Refactor_createfollowship_test) go test -v ./internal/controller/handler

=== RUN   TestCreateFollowship
=== RUN   TestCreateFollowship/CreateFollowship_successfully
=== PAUSE TestCreateFollowship/CreateFollowship_successfully
=== RUN   TestCreateFollowship/InvalidJSON_in_request_body
=== PAUSE TestCreateFollowship/InvalidJSON_in_request_body
=== RUN   TestCreateFollowship/UsecaseError
=== PAUSE TestCreateFollowship/UsecaseError
=== CONT  TestCreateFollowship/CreateFollowship_successfully
=== CONT  TestCreateFollowship/UsecaseError
=== CONT  TestCreateFollowship/InvalidJSON_in_request_body
--- PASS: TestCreateFollowship (0.00s)
    --- PASS: TestCreateFollowship/InvalidJSON_in_request_body (0.00s)
    --- PASS: TestCreateFollowship/CreateFollowship_successfully (0.00s)
    --- PASS: TestCreateFollowship/UsecaseError (0.00s)
PASS
ok      x-clone-backend/internal/controller/handler     0.560s
➜  go git:(feature/#690_Refactor_createfollowship_test) 
```
## Schedule
5/3

Rwatana commented:
@nayuta-ai 
I’ve introduced a fake repository to ensure test isolation for the handler. Please review my PR.

nayuta-ai commented on go/internal/controller/handler/create_followshiop_test.go:
```diff
@@ -0,0 +1,70 @@
+			body:      "invalid json",
+			setupRepo: nil,
+			wantCode:  http.StatusBadRequest,
+			wantBody:  "Request body was invalid.\n",
```
Since we'll be using it in tests as well, let's define the error type and compare it.

nayuta-ai commented on go/internal/controller/handler/create_followshiop_test.go:
```diff
@@ -0,0 +1,70 @@
+
+	for _, tt := range tests {
+		tt := tt
+		ID := "17c78eb1-72c2-43dd-bd96-7eadabeff206"
```
Please define it as a global constant.

Rwatana commented:
@nayuta-ai 
## Topic
I believe I have addressed the points you mentioned. Could you please review my PR?
## Note
I made a mistake during the Git operation, and the PR was automatically closed the moment the commits disappeared.
I'll be more careful next time.

Rwatana commented:
@nayuta-ai 
- I create `error.go`
- I modified to use `ErrInvalidRequestBody` and `ErrCreateFollowship` in create_followship.go.

Please review my PR.

nayuta-ai commented on go/internal/controller/handler/errors.go:
```diff
@@ -0,0 +1,15 @@
+	"errors"
+)
+
+// the error messages when the request body is invalid
```
Please avoid leaving unnecessary comments, and declare errors in the same way as the other variables.

Rwatana commented:
@nayuta-ai 
## Topic
Thank you for reviewing this so late at night, and I’m sorry for asking you to review it so many times. I’ve made the corrections.
Please review my code.
## Question
I have one question: Is the reason we don’t need comments because you can tell what the error is just by looking at the variable name?

Amoonguses1 reviewed (commented):
Use `openapi.CreateFollowshipRequest` and delete `createFollowshipRequestBody`.

ryuju0911 commented:
@Rwatana
You should resolve or reply to all comments.
https://github.com/okuda-seminar/Twitter-Clone/blob/main/README.md#pr--review

Rwatana commented:
@nayuta-ai 

## Topic
I’ve updated the code. Please review my PR.
## File change
- create\_followship.go
Updated the code to use the errors defined in errors.go. Also changed the request/response types to use those defined in the OpenAPI spec, based on the comment from [https://github.com/okuda-seminar/Twitter-Clone/pull/709#pullrequestreview-2813118794](https://github.com/okuda-seminar/Twitter-Clone/pull/709#pullrequestreview-2813118794).

- create\_followship\_test.go
Refactored the test by simplifying it to check only the status code, following the example from [https://github.com/okuda-seminar/Twitter-Clone/pull/717/files#diff-45f4d77f806551add81083f5ea0d36d3a6e326584321dd19d0d4f18040870688](https://github.com/okuda-seminar/Twitter-Clone/pull/717/files#diff-45f4d77f806551add81083f5ea0d36d3a6e326584321dd19d0d4f18040870688). Also unified variable names such as `expected_status` to match the convention used in other tests.

- errors.go
Simplified the error definitions by removing unnecessary comments, referring to the approach in the above PR.

- types.go
Removed the createPostRequestBody definition as instructed in [https://github.com/okuda-seminar/Twitter-Clone/pull/709#pullrequestreview-2813118794](https://github.com/okuda-seminar/Twitter-Clone/pull/709#pullrequestreview-2813118794).

- handlers.go
Removed unused handler functions to simplify the code.

- server.go
Adjusted the handler setup to match the updated OpenAPI-based structure.

## Schedule
https://github.com/okuda-seminar/Twitter-Clone/pull/729 should be merged first due to the "k8s manifests test" issue.

nayuta-ai commented:
/lgtm
/hold
